combat/CombatSystem.cpp

```diff
@@ -238,9 +238,9 @@ void CombatInfo::InitializeObjectVisibility()
         if (empire_id == ALL_EMPIRES)
             continue;
         empire_known_objects[empire_id].Insert(GetEmpireKnownSystem(system->ID(), empire_id));
-        empire_object_visibility[empire_id][system->ID()] = GetUniverse().GetObjectVisibilityByEmpire(empire_id, system->ID());
+        empire_object_visibility[empire_id][system->ID()] = GetUniverse().GetObjectVisibilityByEmpire(system->ID(), empire_id);
         for (int object_id : local_object_ids) {
-            Visibility obj_vis = GetUniverse().GetObjectVisibilityByEmpire(empire_id, object_id);
+            Visibility obj_vis = GetUniverse().GetObjectVisibilityByEmpire(object_id, empire_id);
```

Comment:
this wasn't a major problem worth noting...?

Reply:
It does certainly look like the kind of error that could have really significant consequences, I'll grant you that.

I am pretty sure that my initial description for the PR (which was up for a day or 2 while it was work in progress)  talked about it, and the corresponding commit message still does  (sufficiently I think).  
Not all of the related calls in this section of code had the problem.  Because of the particular use for these faulty cases, it looked to me like the only time they would potentially cause a real problem was when the object ID was no larger than the number of empires, which in the vast majority of galaxy setups is going to mean the issue could only apply to a System, which is never going to be a combat target, so as a practical matter the error was not causing problems.

Is there something about the commit message that you'd like me to change?

Reply:
> Is there something about the commit message that you'd like me to change?

no